//-*-mode:protobuf; mode:font-lock;-*-

syntax="proto3";

import "google/protobuf/descriptor.proto";

package calin;

message SQLOptions
{
  bool dont_store      = 1;
  bool is_key          = 2;
  bool inline_message  = 3;
  string index_name    = 4;
};
 
message FieldOptions
{
  enum Int32StorageType {
    INT_32   = 0;
    INT_16   = 1;
    INT_8    = 2;
  }
      
  string            desc        = 1;
  string            units       = 2;
  Int32StorageType  int32_type  = 3;
  bool              dont_store  = 4;
  SQLOptions        sql         = 100;
};

extend google.protobuf.FieldOptions {
  calin.FieldOptions CFO = 50000;
};

